pkg/yamlpath/path_test.go

```diff
@@ -62,6 +62,24 @@ x:
 		expectedPathErr string
 		focus           bool // if true, run only tests with focus set to true
 	}{
+		{
+			name: "property names",
+			path: "$.store~",
+			expectedStrings: []string{
+				`store
+`,
+			},
+			expectedPathErr: "",
+		},
+		{
+			name: "property names bracket child",
+			path: "$.store['book']~",
+			expectedStrings: []string{
+				`book
+`,
+			},
+			expectedPathErr: "",
+		},
 		{
```

Comment:
I'd like to see a similar test with a list of property names inside the brackets.

Reply:
added